CHANGELOG.md

```diff
@@ -0,0 +1,91 @@
+ - Use VCPKG to build third-party dependencies not included in the code tree, such as SDL2 and zlib (#137, #430, #469, #512).
+ - Set up automated build and testing using Github Actions for Linux (clang/gcc), Windows (MSVC) and MacOS (apple-clang, x64 & arm64) (#14, #28, #29, #45, #50, #60, #123, #133, #139, #155, #166, #177, #182)
+ - Modernize build system: require CMake 3.20, use present-day CMake practices (#20, #63, #77, #86, #99, #104, #127, #131, #144, #220, #225, #227, #235, #256, #318, #340, #400, #407, #422, #460, #463, #482, #504)
+ - Ensure compatibility with popular compilers, convert misc. pieces of code to modern C++ (#17, #20, #23, #40, #54, #58, #70, #108, #304, #328, #344, #358, #388, #405, #416)
+ - Mark relevant variables and functions as static (#185)
+ - Update the module system: isolate code for independant sub-modules in separate directories and CMake targets (#76, #89, #91, #195, #367, #374, #448, #470, #502)
+ - Document build procedure and usage on all platforms (#116, #118, #119, #124, #147, #154, #170, #174, #175, #184, #190, #204, #210, #211, #222, #259, #261, #270, #285, #297, #384, #413, #435)
+ - Unifiy Windows & Unix command line arguments: --foo, -foo and +foo is accepted on any OS (#199)
+ - Refactor CleanupStr in misc/pstring.cpp (#207)
+ - Change non-specific types (e.g. unsigned int) to their stdint specified counterparts (e.g. `uint32_t`) (#268, #464)
+ - Support case-insensitive loading of content files (#294)
+ - Build TCP/IP and PXO multiplayer modules (#296, #302, #420, #424, #432)
+ - Refactor renderer: move non-OpenGL specific code to a base file, abstract triangle rendering code (#315, #323, #336)
+ - Save screenshots in PNG format instead of the legacy TGA format (#325, #348)
+ - Bump max cells value to increate render distance for outdoor terrain (#338)
+ - Use chrono C++11 steady_clock for time measurement (#394)
+ - Enable real-time library memory functions by default (#436, #445, #456)
+ - Change ``__LINUX__` defines to `POSIX` for general OS checks (#492)
+
+
+### Fixed
+ - Cleanup warnings issued by modern compilers (#57, #74, #82, #87, #89, #109, #117, #218, #349, #387, #428, #434, #478, #507)
+ - Replace sprintf by snprintf(#78, #252, #266)
+ - Add object unlink sanity check (#140)
+ - Use a static background image in the main menu if the video file is not present (#152)
+ - Add virtual default constructors on abstract classes (#219)
+ - Fix cursor not being grabbed on macOS (#223)
+ - Fix detection of mercenary installation, where hogfiles are not the same for CD and digital installation (#260)
+ - Fix some out-of-bounds array access (#263)
+ - Fix for Level 5 script;  Make CustomText2 centered on Y axis (#307)
+ - Fix a crash when calling `quit` in the server console in dedicated server mode (#370)
+ - Resolves an issue where hardware cursor was still visible in release build despite window mode being disabled for release builds. (#371)
```

Comment:
Is window mode still disabled for release? With SDL2 and FBO, it should be stable enough now I think.

Reply:
Good point, it is still disabled indeed. We need testing to see if it's safe to enable now

Reply:
Game compiled as release can run in windowed mode but crashes after movie playback, probably due to an MVE error. I'll see if I can fix that first

Reply:
I don't get any crashes in Release windowed mode. Is it a specific movie?

Reply:
I crash when playing `level1.mve`, entering level 1 for the first time. If needed I can try to give a more specific reproducer @winterheart were you able to replicate?

Reply:
see #523